docs/source/nlp/dialogue.rst

```diff
@@ -0,0 +1,128 @@
+| Sequence Generation                                      | Schema Guided Dialogue Generation| Dialogue GPT Generation Model                                                    |
++                                                          +----------------------------------+----------------------------------------------------------------------------------+
+|                                                          | MS Marco NLGen                   | Dialogue S2S Generation Model                                                    |
++----------------------------------------------------------+----------------------------------+----------------------------------------------------------------------------------+
+
+**Configuration** 
+
+Example of model configuration file for training the model can be found at: `NeMo/examples/nlp/dialogue/conf/dialogue_config.yaml <https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/dialogue/conf/dialogue_config.yaml>`__.
+
+Because the Dialogue module contains a wide variety of models and tasks, there are a large number of configuration parameters to adjust (some of which only applies to some models/some tasks)
+
+In the configuration file, define the parameters of the training and the model, although most of the default values will work well.
+For various task-model combination, only a restricted set of config args will apply. Please read the configuration file for comments on which config args you would need for each model and task.
+
+The configuration can be roughly grouped into a few categories:
+
+- Parameters that describe the training process, such as how many gpus to use: **trainer**
+- Parameters that describe the model: **model**
+- Parameters that describe optimization: **model.optim**
+- Parameters that describe the task: **model.dataset**
+- Parameters that describe the dataloaders: **model.train_ds**, **model.validation_ds**, **model.test_ds**,
+- Parameters that describe the training experiment manager that log training process: **exp_manager**
+
+
+Arguments that very commonly need to be edited for all models and tasks
+
+- :code:`do_training`: perform training or only testing
+- :code:`trainer.devices`: number of GPUs (int) or list of GPUs e.g. [0, 1, 3]
+- :code:`model.dataset.task`: Task to work on [sgd, assistant, zero_shot, ms_marco, sgd_generation, design, mellon_qa]
+- :code:`model.dataset.data_dir`: the dataset directory
+- :code:`model.dataset.dialogues_example_dir`: the directory to store prediction files
+- :code:`model.dataset.debug_mode`: whether to run in debug mode with a very small number of samples [True, False]
+- :code:`model.language_model.pretrained_model_name`: language model to use, which causes different Dialogue Models to be loaded (see dialogue.py for details) [t5-base, facebook/bart-large, bert-base-uncased, sentence-transformers/all-MiniLM-L6-v2, microsoft/DialoGPT-small]
```

Comment:
For the pretrained language models, here you point to dialogue.py for details. dialogue.py then points to the example config for details. The comment in the config doesn't completely make things clear. What determines whether a model is usable in your framework? Will any HuggingFace model work? Will any NGC model work (I'm asking that rhetorically - I'm sure the answer is no, but how can a user figure out which NGC models are usable?) For the models that are usable here, is every pretrained language model compatible with every task? (Again I'm sure the answer is no, but it would be helpful if you could spell that out here)

Reply:
Thanks for catching this! I have now updated the docs!